block-layer/codecs/bitcoin.md

```diff
@@ -0,0 +1,220 @@
+
+The block graph encodes the transactions in a sequential list, prefixed by a "compact size" integer. See below for how this variable integer is encoded. This integer dictates the number of transactions to follow.
+
+Each individual transaction can be described with the following IPLD Schema.
+
+```ipldsch
+type Transaction struct {
+  version Int
+  segwit optional Bool (implicit "false")
+  vin [TxIn]                       # prefixed by a compact size int
+  vout [TxOut]                     # prefixed by a compact size int
+  scriptwitness optional [Witness] # same number of elements as `vin`
+  locktime Int
+}
+
+type TxIn struct {
+  prevOut OutPoint
+  scriptSig Bytes # unbounded vec<char>
+  sequence Int
+}
+
+type OutPoint struct {
+  hash Bytes # 256-bits
+  n Int
+}
+
+type TxOut struct {
+  value Int                # int64
+  scriptPubKey MaybePubKey # unbounded vec<char>
+}
+
+type Witness bytes # unbounced vec<bytes>
+```
+
+Notes:
+
+* Each of the lists are prefixed by their size, using the compact size format (see below). There must be exactly that number of elements in that list.
+* Each `vec<char>` is prefixed by its size, using the compact size format. There must be exactly that number of bytes. There may be zero or more bytes in the byte array.
+
+Decoding a Transaction into this structure requires reading the following elements in order:
+
+* `version`: a signed 32-bit integer
+* `segwit`: is implicit and `false` for all block graphs prior to the SegWit soft fork, which occurred at a height of 481,824. After this height, the two bytes following `version` are inspected, if they are equal to `[0x0, 0x1]`, the bytes are consumed and `segwit` is `true`. If the bytes are not exactly these values, `segwit` is false, and the two bytes instead form the begining of `vin` (the first byte of `vin` is part of the compact size integer, and as `vin` must contain one or more elements, it cannot be `0x00`, hence the reliability of the `segwit` flag maintaining backward-compatibility).
+* `vin`: one or more elements, prefixed by a compact size int, then, for each element up to the size:
+  * `hash`: an unsigned 256-bit integer / a 32-byte binary string, the OutPoint transaction ID hash identifying the source transaction for the coins
```

Comment:
This goes together with the endianness discussion above: being an integer and a string at the same time can't be a thing.